lib/widgets/store.dart

```diff
@@ -246,3 +249,36 @@ class LoadingPage extends StatelessWidget {
     return const Center(child: CircularProgressIndicator());
   }
 }
+
+/// A [State] that uses the ambient [PerAccountStore].
+///
+/// The ambient [PerAccountStore] can be replaced in some circumstances,
+/// such as when an event queue expires. See [PerAccountStoreWidget.of].
+/// When that happens, stateful widgets should
+/// - remove listeners on the old [PerAccountStore], and
+/// - add listeners on the new one.
+/// Use this mixin, overriding [onNewStore], to do that concisely.
+// TODO(#185): Until #185, I think [PerAccountStoreWidget.of] never actually
+//   returns a different [PerAccountStore] from one call to the next.
+//   But it will, and when it does, we want our [StatefulWidgets] to handle it.
+mixin PerAccountStoreAwareStateMixin<T extends StatefulWidget> on State<T> {
+  PerAccountStore? _store;
+
+  /// Called in the first [State.didChangeDependencies] call, and afterward
+  /// when [PerAccountStoreWidget.of] returns a different store than last time.
+  ///
+  /// In this, remove any listeners on the old store
+  /// and add them on the new store.
+  void onNewStore();
```

Comment:
Here's a version which in particular tries to add a succinct summary:
```suggestion
  /// Called when there is a new ambient [PerAccountStore].
  ///
  /// Specifically this is called when this element is first inserted into the tree
  /// (so that it has an ambient [PerAccountStore] for the first time),
  /// and again whenever dependencies change so that [PerAccountStoreWidget.of]
  /// would return a different store from previously.
  ///
  /// In this, remove any listeners on the old store
  /// and add them on the new store.
  void onNewStore();
```

Reply:
Thanks!